bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,74 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(minutes=10.0)
+    async def completer_task(self) -> None:
+        """
+        Give members who have completed all 50 AoC stars completionist role.
+
+        Runs on a schedule, as defined in the task.loop decorator.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+        if completionist_role is None:
+            log.warning("Could not find the AoC completionist role, cancelling sync task.")
+            self.completer_task.cancel()
+            return
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
+        }
+
+        try:
+            leaderboard = await _helpers.fetch_leaderboard()
+        except _helpers.FetchingLeaderboardFailedError:
+            await self.bot.send_log("Unable to fetch AoC leaderboard during role sync.")
+            return
+
+        placement_leaderboard = json.loads(leaderboard["placement_leaderboard"])
+
+        for member_aoc_info in placement_leaderboard.values():
+            if not member_aoc_info["stars"] == 50:
+                # Only give the role to people who have completed all 50 stars
+                continue
+
+            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            if not member_id:
+                continue
+
+            member = await members.get_or_fetch_member(guild, member_id)
+            if member is None:
+                continue
+
+            if completionist_role in member.roles:
+                continue
+
+            await members.handle_role_change(member, member.add_roles, completionist_role)
+
     @commands.group(name="adventofcode", aliases=("aoc",))
     @whitelist_override(channels=AOC_WHITELIST)
     async def adventofcode_group(self, ctx: commands.Context) -> None:
         """All of the Advent of Code commands."""
         if not ctx.invoked_subcommand:
             await invoke_help_command(ctx)
 
+    @with_role(Roles.admins)
+    @adventofcode_group.command(
+        name="block",
+        brief="Block a user from getting the completer role.",
+    )
+    async def block_from_role(self, ctx: commands.Context, member: discord.Member) -> None:
+        """Block the given member from receiving the AoC completionist role, removing it from them if needed."""
+        completionist_role = ctx.guild.get_role(Roles.aoc_completionist)
+        if completionist_role in member.roles:
+            await member.remove_roles(completionist_role)
+
+        await self.completionist_block_list.set(member.id, "sentinel")
```

Comment:
This isn't getting read anywhere any more. The task needs to avoid adding a role if the user is in this.

Reply:
Yes, this was dumb of me to remove, I shall add back in the contains check.

Reply:
Added back https://github.com/python-discord/sir-lancebot/pull/991/commits/882de8b58a5cf60959fc861938c677f0448355d2#diff-89718a16af42ae2ad5c407a19adc6622a45182459c2adda664b3bd35d8fbf193R105-R106